src/workerd/api/streams/internal.c++

```diff
@@ -290,12 +290,14 @@ public:
     // isolate.
     if (!wasRead) {
       auto msg =
-          "Your worker called response.clone(), but did not read the body of both clones. "
-          "This is wasteful, as it forces the system to buffer the entire response body in memory, "
-          "rather than streaming it through. This may cause your worker to be unexpectedly "
+          "Your worker created multiple branches of a single stream (for instance, by calling "
+          "`response.clone()` or `request.clone()`) but did not read the body of both branches. "
+          "This is wasteful, as it forces the system to buffer the entire stream of data in "
+          "memory, rather than streaming it through. This may cause your worker to be unexpectedly "
           "terminated for going over the memory limit. If you only meant to copy the "
-          "response headers and metadata (e.g. in order to be able to modify them), "
-          "use `new Response(response.body, response)` instead.";
+          "request or response headers and metadata (e.g. in order to be able to modify them), "
+          "use the appropriate constructors instead (for instance, `new Response(response.body, "
```

Comment:
We should maybe spell out exactly how to copy a Request, especially as it is different from the Response constructor in how the body is treated.

`new Request(request)`

Reply:
^ supportive of this.